app/views/pages/home.html.erb

```diff
@@ -174,7 +174,13 @@
               <br><br>
               <b><%= p.challenge.name %></b><br>
               <% if p.challenge.postfrequency == 0 %>
-                <span class="streakNeutral">No post frequency. Meet the submission count requirement by: <%= p.next_submission_date.strftime("%a, %d %b %Y") %></span>
+                <% if p.challenge.end_date == Time.now.utc.tomorrow.beginning_of_day && p.challenge.challenge_entries.count < p.challenge.badge_maps.first.required_score %>
+                  <span class="streakDanger">You must meet the submission count requirement the end of the day: <%= p.challenge.end_date.strftime("%a, %d %b %Y") %></span>

```

Comment:
generally speaking, non-post frequency challenges run long enough that we might want to consider more granular warnings if we have any at all.
what i mean is, for projects that last months long, a last minute "you have until tomorrow to submit everything" might be woefully insufficient
im going to make a case that this is something we probably want to revisit when the challenges are reworked; as to attend to this issue means a bit more work that will get replaced anyway.

there is also an issue with using Time.now.utc.beginning_of_day... you should probably use Time.now.utc.to_date, as to ensure date vs date rather than date vs datetime comparison.

Reply:
Wouldn't your reminder be covered by the case on line 182? How much do we _really_ need to remind the user to fulfill challenge requirements?

```
<% else %>
  <span class="streakNeutral">No post frequency. Meet the submission count requirement by: <%=  p.challenge.end_date.strftime("%a, %d %b %Y") %></span>
<% end %>

```

Reply:
I figure for something like this, it's sort of like, 50% of the days elapsed (rounded down), show a yellow (orange? yellow's already streak neutral.) warning saying "the deadline is approaching", you need to meet the submission count soon! Then, when 14% (or 1 day, whichever is greater) before the deadline is there, you show the same warning in red, with the statement changing to "the end of the day" on the last day. That way, there's a touch more granularity for projects that run anywhere from a year to a month to just a week.
That's the most dev-simple way to do it, I think, without committing too much effort before the challenge overhaul. I just think the "last day" reminder might not be that useful for the challenge if it spans longer than two weeks.